### PR TITLE
Update TS config to have an isolated TS environment for woo admin

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

When we run `pnpm nx ts:check woocommerce-admin,` it also checks the woo packages. But everything in the monorepo is supposed to be isolated. So I made the following changes to have an isolated environment:

- Change the admin TS config and removed client TS config 
- Update the packages TS config to output declaration files 
- Move `plugins/woocommerce-admin/typings` to `plugins/woocommerce-admin/client/typings` to load them correctly.
- Disable `ForkTsCheckerWebpackPlugin` for now

### How to test the changes in this Pull Request:

1. Run `pnpm nx build woocommerce-admin` to build packages
2. `packages/js/data/build-types` should be generated
3. Run `pnpm nx ts:check woocommerce-admin`
4. Should see many client TS errors.
5. Run `pnpm nx clean woocommerce-admin & pnpm nx ts:check woocommerce-admin`
6. Should see some TS errors like `Cannot find module '@woocommerce/data' or its corresponding type declarations.` 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/webpack.config.js [hunk 1]
```diff
@@ -133,7 +133,8 @@ const webpackConfig = {
 	},
 	plugins: [
 		...styleConfig.plugins,
-		new ForkTsCheckerWebpackPlugin(),
+		// Disable TS checker for now as it will block all development until all type inconsistencies are fixed
+		// new ForkTsCheckerWebpackPlugin(),
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/webpack.config.js:144]: Just dropping a reference to https://github.com/woocommerce/woocommerce/issues/32581. We should remember to re-enable it after TS errors are fixed

**Review by ilyasfoo — COMMENTED** [review]
Thanks for making this change, @chihsuan!

Something I don't understand, if we're isolating the check, why are we still checking packages' TS when we run `pnpm nx ts:check woocommerce-admin`?

It seems our packages have different configuration in `tsconfig.json`, such as some of them not having `declarationDir`, the `outDir` in `admin-e2e-tests` and `api` has different values, and so on:

image

Do you think it's appropriate to configure our packages to have similar configuration? I feel the difference may make things slightly harder to standardize, such as `clean` scripts have to look for different folders to remove build artifacts. Though I'm not sure if the difference are related to how the package is consumed, may have some implications if we're not aware.

**Comment by chihsuan** [comment]
> Something I don't understand, if we're isolating the check, why are we still checking packages' TS when we run pnpm nx ts:check woocommerce-admin?

@ilyasfoo ` pnpm nx ts:check woocommerce-admin` should not check packages TS. 

For example, if you try to break the package type by removing [this `React.HTMLAttributes< HTMLSpanElement >`](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/badge/index.tsx#L8) from `BadgeProps`. When you run `pnpm nx ts:check woocommerce-admin`, you should not see `packages/js/components/src` type error. But you can see the error when you run `pnpm nx build @woocommerce/components`.

However, we need to build packages before running admin `ts:check` because packages are our dependencies. We need their type declaration like our other dependencies in node_modules.

> It seems our packages have different configuration in tsconfig.json, such as some of them not having declarationDir, the outDir in admin-e2e-tests and api has different values, and so on:

Some of them are not having `declarationDir` because not all packages contain `.ts` files. For instance, `@woocommerce/notices` is still a pure JS package, so we don't need to output TS declaration.

The outDir in admin-e2e-tests and api has different values because `api` is not originated from woocommerce-admin repo and is maintained by other teams.

> Do you think it's appropriate to configure our packages to have similar configuration? I feel the difference may make things slightly harder to standardize, such as clean scripts have to look for different folders to remove build artifacts. Though I'm not sure if the difference are related to how the package is consumed, may have some implications if we're not aware.

I think it's okay that each package has different configuration because they are isolated. But yes, I agree we could standardize them to make it better for management. I'll create another issue for this.

For the clean script, we use `pnpm run:packages -- clean --parallel` so it actually run the "clean" script that defined in each package to remove their build artifacts correctly.

**Comment by ilyasfoo** [comment]
> However, we need to build packages before running admin ts:check because packages are our dependencies.

Ah, of course. I missed this fact. Thanks for clearing that up!

> Some of them are not having declarationDir because not all packages contain .ts files. For instance, @woocommerce/notices is still a pure JS package, so we don't need to output TS declaration.

Got that!

>I think it's okay that each package has different configuration because they are isolated. But yes, I agree we could standardize them to make it better for management. I'll create another issue for this.
For the clean script, we use pnpm run:packages -- clean --parallel so it actually run the "clean" script that defined in each package to remove their build artifacts correctly.

Thanks! I agree with your point that each project would have its own clean script, my point was pretty weak, I think the best benefit for standardized config would be the long term maintainability & easier to reason when looking across packages.

Thanks for letting me pick your brain on this! 😄

**Review by ilyasfoo — DISMISSED** [review]
Thanks for this, @chihsuan! Tested well and LGTM 🚢

**Pull request merged**